### PR TITLE
tools/ut: accelerate the build process and scatter test

### PR DESCRIPTION
**Pull request by HuSharp**
### What problem does this PR solve?

accelerate pd-ut, for now all tests can be executed in 5min

Issue Number: Ref #7969

### What is changed and how does it work?

- enable `race`
- divide unit test/tests/tso integration test into 2 parts respectively
- accelerate the build process by concurrency
- add `junitfile` in ci to indicate test time

image

check junitfile

image

```commit-message
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

### Release note

```release-note
None.
```

### CODE REVIEW

File: .github/workflows/pd-tests.yaml [hunk 1]
```diff
@@ -27,19 +27,27 @@ jobs:
...
+            name: 'Tests(others)'
           - worker_id: 6
-            name: 'MicroService Integration Test'
+            name: 'Tools Test'
+          - worker_id: 7
+            name: 'Client Integration Test'
+          - worker_id: 8
+            name: 'TSO Integration Test'
+          - worker_id: 9
+            name: 'MicroService Integration(Core)'
```
Comment (rleungx) [inline .github/workflows/pd-tests.yaml:46]: What does the `Core` here mean?

File: tests/testutil.go [hunk 2]
```diff
@@ -283,44 +283,52 @@ func MustReportBuckets(re *require.Assertions, cluster *TestCluster, regionID ui
...
 		t:        t,
 		opts:     opts,
-		clusters: make(map[mode]*TestCluster),
+		clusters: make(map[SchedulerMode]*TestCluster),
 		cancels:  make([]context.CancelFunc, 0),
 	}
 }
 
-// RunTestInTwoModes is to run test in two modes.
 func (s *SchedulingTestEnvironment) RunTestInTwoModes(test func(*TestCluster)) {
```
Comment (rleungx) [inline tests/testutil.go:313]: The name should be changed?

Comment (HuSharp) [reply tests/testutil.go]: changed to `RunTestBasedOnMode`, PTAL thx!

File: .github/workflows/pd-tests.yaml (hunk 1 above)
Comment (HuSharp) [reply .github/workflows/pd-tests.yaml]: Oh, I forgot to change that, changed to `Unit Test(X)`
Categorization based only on test time, which is determined by the eye ...Do you have any good ideas?

File: scripts/ci-subtask.sh [hunk 3]
```diff
@@ -3,34 +3,51 @@
...
+ROOT_PATH_JUNITFILE=$(pwd)/junitfile
 # Currently, we only have 3 integration tests, so we can hardcode the task index.
 integrations_dir=$(pwd)/tests/integrations
 
 case $1 in
     1)
-        # unit tests ignore `tests`
-        ./bin/pd-ut run --race --ignore tests --coverprofile $ROOT_PATH_COV || exit 1
+        # unit tests ignore `tests`, `server`, `schedule`,`utils` and `encryption`
+        ./bin/pd-ut run --ignore tests,server,pkg/schedule,pkg/utils,pkg/encryption --race --coverprofile $ROOT_PATH_COV --junitfile $ROOT_PATH_JUNITFILE || exit 1
```
Comment (rleungx) [inline scripts/ci-subtask.sh:13]: What if these packages are deleted or renamed?

Comment (HuSharp) [reply scripts/ci-subtask.sh:13]: It may be necessary to manually change the code,when noticing abnormal `test time`/`test codecov`

Comment (HuSharp) [reply scripts/ci-subtask.sh:13]: - codev can reveal missing tests
- Redundant tests can be reacted to at test time

**Review by rleungx — APPROVED** [review]

**Comment by wuhuizuo** [comment]
/hold
/assign @purelind 
Please hold the merging with the CI updates.

**Review by lhy1024 — APPROVED** [review]

**Review by purelind — APPROVED** [review]
/lgtm

**Pull request merged**